File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/dashboard/DashboardRouter.java
package com.tai.androidtai.modules.dashboard;

import android.content.Intent;

import com.tai.androidtai.domain.bean.MealBean;
import com.tai.androidtai.modules.mealDetails.MealDetailsActivity;

import androidx.appcompat.app.AppCompatActivity;

public class DashboardRouter implements DashboardContract.Router {

    private final AppCompatActivity mActivity;

    public DashboardRouter(AppCompatActivity activity) {
        mActivity = activity;
    }

    @Override
    public void goToMealDetails(MealBean mealBean, String price) {
        Intent intent = new Intent(mActivity, MealDetailsActivity.class);
        intent.putExtra("meal", mealBean);
        intent.putExtra("price", price);
        mActivity.startActivity(intent);
    }
}

File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/dashboard/DashboardRouter.kt
package com.tai.androidtai.modules.dashboard

import android.content.Intent
import com.tai.androidtai.domain.bean.ResultBean
import com.tai.androidtai.modules.core.BaseRouter
import com.tai.androidtai.modules.userDetails.UserDetailsActivity
import com.tai.androidtai.modules.userDetails.UserDetailsActivity.Companion.USER_ID

class DashboardRouter : BaseRouter(), DashboardContract.Router {

    override fun goToUserDetails(resultBean: ResultBean, view: DashboardContract.View?) {
        val intent = Intent(getActivity(view!!), UserDetailsActivity::class.java)
        intent.putExtra(USER_ID, resultBean.getId())
        getActivity(view)?.startActivity(intent)
    }
}
